lib/CL/clEnqueueMapImage.c

```diff
@@ -139,40 +146,34 @@ CL_API_SUFFIX__VERSION_1_0
   DL_APPEND (image->mappings, mapping_info);
   POCL_UNLOCK_OBJ (image);
 
-  errcode = pocl_create_command (&cmd, command_queue, CL_COMMAND_MAP_IMAGE, 
-                                 event, num_events_in_wait_list, 
-                                 event_wait_list);
-  if (errcode != CL_SUCCESS)
-    goto ERROR;
-      
-  
   cmd->command.map.buffer = image;
   cmd->command.map.mapping = mapping_info;
   pocl_command_enqueue(command_queue, cmd);
-  
+
+  *image_row_pitch = image->image_row_pitch;
+  if (image_slice_pitch)
+    *image_slice_pitch = image->image_slice_pitch;
+
   if (blocking_map)
     {
       POname(clFinish) (command_queue);
     }
-  
-  *image_row_pitch = image->image_row_pitch;
-  if (image_slice_pitch)
-    *image_slice_pitch = image->image_slice_pitch;
 
+  if (event)
+    *event = _event;
   if (errcode_ret != NULL)
     (*errcode_ret) = CL_SUCCESS;
 
   return map;
- 
- ERROR:
-  POCL_MEM_FREE(map);
+
+ERROR:
+  POCL_MEM_FREE(_event);
+  // POCL_MEM_FREE(map); // NOPE! TODO FIXME this should be a call to unmap_mem
```

Comment:
Actually, with the reordered code map should never be != NULL here, since there is no point of failure after it has been assigned a non-NULL value. I can replace that line with an assert or just remove it.